controlpanel/frontend/jinja2/base.html

```diff
@@ -81,13 +81,13 @@
         "active": page_name == "warehouse-datasource-list"
       },
       {
-        "hide": not request.user or not request.user.userapps or not request.user.userapps.filter(is_admin=True).exists(),
+        "hide": not request.user.users3buckets.filter(s3bucket__is_data_warehouse=False).exists(),
```

Comment:
Looks like quite a bit of low level/logic for a template. But don't worry too much, just an observation, we can improve later.

Reply:
Don't be afraid of logic in templates 😉

Reply:
you know I'm afraid of my shadow Andy 😁